src/cpdetect/compiler_detector/heuristics/pe_heuristics.cpp

```diff
@@ -327,6 +338,79 @@ void PeHeuristics::getVisualBasicHeuristics()
 	}
 }
 
+int32_t PeHeuristics::get_int32_unaligned(const uint8_t * codePtr)
+{
+	const int32_t * codePtrInt32 = reinterpret_cast<const int32_t *>(codePtr);
+
+    return codePtrInt32[0];
+}
+
+/**
+ * Parses the code, follows NOPs or JMPs
+ */
+const uint8_t * PeHeuristics::skip_NOP_JMP8_JMP32(const uint8_t * codeBegin, const uint8_t * codePtr, const uint8_t * codeEnd, size_t maxCount)
+{
+	while(codeBegin <= codePtr && codePtr < codeEnd && maxCount > 0)
+	{
+		ssize_t movePtrBy = 0;
+
+		switch (codePtr[0])
+		{
+			case 0x90:  // NOP, move by 1 byte
+				movePtrBy = 1;
+				break;
+
+			case 0xEB:	// JMP rel8
+				if((codePtr + 2) > codeEnd || codePtr[1] == 0x80)
+					return nullptr;
+				movePtrBy = static_cast<int8_t>(codePtr[1]);
+				break;
+
+			case 0xE9:
+				if ((codePtr + 5) > codeEnd || get_int32_unaligned(codePtr+1) == 0x80000000)
+					return nullptr;
+				movePtrBy = get_int32_unaligned(codePtr + 1);
+				break;
+
+			default:
+				break;
+		}
+
+		// If no increment, it means there was no NOP/JMP and we are done
+		if(movePtrBy == 0)
+			break;
+
+		// Check whether we got into the code range
+		codePtr = codePtr + movePtrBy;
+		maxCount--;
+	}
+	return codePtr;
+}
+
+/**
+ * Detect header heuristics
+ */
+void PeHeuristics::getHeaderStyleHeuristics()
+{
+	const auto &content = search.getPlainString();
+
+	// Must have at least IMAGE_DOS_HEADER
+	if(content.length() > 0x40)
+	{
+		const char * e_cblp = content.c_str() + 0x02;
+
+		for (size_t i = 0; i < headerStyles.size(); i++)
+		{
+			if(!memcmp(e_cblp, headerStyles[i].headerWords, sizeof(headerStyles[i].headerWords)))
```

Comment:
Two notes concerning `memcmp`:
* Please, qualify it with `std::` (for the same reasons as [here](https://cs-blog.petrzemek.net/2015-07-18-zajimavost-z-cpp-size_t-vs-std-size_t)).
* I think the include of `cstring` is missing in the file (it provides `std::memcmp`).